File: 30daysOFcode_Go/day 28.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "strconv"
    "strings"
    "sort"
)



func main() {
    reader := bufio.NewReaderSize(os.Stdin, 1024 * 1024)

    NTemp, err := strconv.ParseInt(readLine(reader), 10, 64)
    checkError(err)
    N := int32(NTemp)
    var names []string 
    for NItr := 0; NItr < int(N); NItr++ {
        firstNameEmailID := strings.Split(readLine(reader), " ")

        firstName := firstNameEmailID[0]

        emailID := firstNameEmailID[1]
        i := 0; n := len(emailID)
        for i < n && emailID[i] != '@'{
            i++;
        }

        if(i+6 < n && emailID[i+1:i+6] == "gmail" && i+10 <= n && emailID[i+7:i+10] == "com"){
            names = append(names, firstName)
        }
    }
    sort.Strings(names)
    for i := 0; i < len(names); i++{
        fmt.Println(names[i])
    }
}

func readLine(reader *bufio.Reader) string {
    str, _, err := reader.ReadLine()
    if err == io.EOF {
        return ""
    }

    return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
    if err != nil {
        panic(err)
    }
}

File: 30daysOFcode_TypeScript/day 28.ts
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString: string = '';
let currentLine: number = 0;

process.stdin.on('data', inputStdin => {
  inputString += inputStdin;
});

process.stdin.on('end', _ => {
  inputString = inputString.replace(/\s*$/, '')
    .split('\n')
    .map(str => str.replace(/\s*$/, ''));

  main();
});

function readLine() {
  return inputString[currentLine++];
}



function main() {
  const N = parseInt(readLine(), 10);

  const nameArr: string[] = [];
  for (let NItr = 0; NItr < N; NItr++) {
    const firstNameEmailID: string[] = readLine().split(' ');

    const firstName: string = firstNameEmailID[0];

    const emailID: string = firstNameEmailID[1];

    const gmailRegex: RegExp = new RegExp('^[a-zA-Z0-9_.+-]+@gmail.[a-zA-Z0-9-.]+$')

    if (emailID.match(gmailRegex)) {
      nameArr.push(firstName)
    }
  }

  nameArr.sort();

  for (const name of nameArr) {
    console.log(name);
  }

}